driver/pparse.ml

```diff
@@ -189,16 +189,19 @@ let check_loc_ghost (type a) (kind : a ast_kind) (ast : a) ~inputfile =
     in
     Ast_invariants.check_loc_ghost meth ast ~source_contents
 
-let file_aux ~tool_name ~sourcefile inputfile (type a) parse_fun invariant_fun
-             (kind : a ast_kind) : a =
-  let ast =
+type 'a ast_result = { ast : 'a; source_file : string }
+
+let file_aux ~tool_name ~source_file inputfile (type a) parse_fun invariant_fun
```

Comment:
I don't expect those changes to be still here after switching to `Location.input_name` for conveying the original input name, but the style guide  is to keep original style like `~sourcefile` rather than rewrite unrelated code for uniformity reason.